File: DukeVineland.Dtos/BookEditorDtos/AddBookRequestDto.cs
namespace DukeVineland.Dtos.BookEditorDtos
{
    public class AddBookRequestDto
    {
        public DateTime Date { get; set; }

        public string Author { get; set; } = string.Empty;

        public string Title { get; set; } = string.Empty;

        public ushort Pages { get; set; }

        public string Note { get; set; } = string.Empty;

        public string Nationality { get; set; } = string.Empty;

        public string OriginalLanguage { get; set; } = string.Empty;

        public string ImageUrl { get; set; } = string.Empty;

        public List<string> Tags { get; set; } = [];

        public string UserId { get; set; } = string.Empty;

        public string Format { get; set; } = string.Empty;
    }
}

File: dukevineland.client/src/app/forms/models/add-book-request-dto.ts
export interface IAddBookRequestDto {

  date: Date;
  author: string;
  title: string;
  pages: number;
  note: string;
  nationality: string;
  originalLanguage: string;
  imageUrl: string;

  tags: string[];
  format: string;

  userId: string;
}
